src/java.base/share/classes/java/nio/file/Path.java

```diff
@@ -248,6 +249,63 @@ public static Path of(URI uri) {
      */
     Path getFileName();
 
+    /**
+     * Returns the file name extension of this path as a {@code String}.
```

Comment:
The latest proposal looks good but I wonder if we can improve on "Returns the file name extension of this path".  What would you think of "Returns the file extension of this path's file name" ?

Reply:
That sounds reasonable.

Reply:
So changed in 66cb4a42b15b2db0427125e404684060b59b6f39.